typedargs/doc_parser.py

```diff
@@ -321,14 +321,83 @@ def wrap_and_format(self, width=None, include_params=False, include_return=False
         return out.getvalue()
 
 
+def _parse_param_validators(param_desc):
+    """Get validators from parameter description.
+
+    Validators should be specified in braces. Arguments of validator should be in brackets.
+    Example of param_desc value:
+
+        {nonnegative, range(1, 5), list(['a', 'b'], istrue(True))} Parameter descriptiom
+
+    Validator arguments in parenthesis should be valid python code.
+    Allowed validator argument types are: str, int, bool, float, None, list
+
+    Args:
+        param_desc (str): parameter description
+
+    Returns:
+          List[Tuple[str, list]]: list of validators
+    """
+
+    param_desc = param_desc.strip()
+    if not param_desc.startswith('{'):
+        return []
+
+    validators_string, _br, _ = param_desc.partition('}')
+    if not _br:
+        raise ValidationError('Malformed validators notation. Closing brace missed.', param_desc=param_desc)
+
+    validators_string = validators_string.lstrip('{')
+
+    result_list = []
+    while validators_string:
```

Comment:
While this logic may work, it is complicated and would benefit from a refactor.

I would recommend:

```python

def iter_validators(validators_string):
    """Iterate over each complete validator."""

    while validators_string:
         # do whatever you need to yield each individual validator
         yield the_next_validator_from_string
```

Then inside `parse_param_validators` you can do:

```python
for validator in iter_validators(validators_string):
    # Do whatever
```

That way you don't mix the logic to get the next validator with the logic to parse the given validator.

Reply:
make sense. Thanks. done - 052d64b86d2